### ISSUE TITLE
Global externals can leak into es bundle when using multiple targets

### ISSUE THREAD

**Issue by such** [rollup/rollup#875]
Hello,

I created a minimal project that reproduces the issue : https://github.com/such/test-rollup.

My project b depends on a **class** from project a:

```
import { A } from 'a';

export default function() {
  const myA = new A();
  myA.print();
  console.log('In module b');
}
```

The output file from building an es bundle of b together with an umd bundle is:

```
import { A } from 'a';

function index() {
  const myA = new a.A();
                 ^^^
  myA.print();
  console.log('In module b');
}

export default index;
```

even though when we only build the es bundle we get the correct output:

```
import { A } from 'a';

function index() {
  const myA = new A();
  myA.print();
  console.log('In module b');
}

export default index;
```

Let me know if you need more information.

**Comment by Rich-Harris** [comment]
Thanks for the test repo, very helpful – have opened #885

**Comment by Rich-Harris** [comment]
Fixed in 0.34.11
